src/bot.ts

```diff
@@ -51,35 +62,56 @@ const onIssueComment = async (state: State, context: IssueCommentCreatedContext,
 };
 
 const main = (bot: Probot) => {
-  const allowedTipRequesters = JSON.parse(envVar("ALLOWED_USERS")) as unknown[];
-  if (!Array.isArray(allowedTipRequesters)) {
-    throw new Error("$ALLOWED_USERS needs to be an array");
-  }
-
-  const seedOfTipperAccount = envVar("ACCOUNT_SEED");
-
-  const state = { allowedTipRequesters, seedOfTipperAccount, bot };
+  const state: State = {
+    bot,
+    allowedGitHubOrg: envVar("APPROVERS_GH_ORG"),
+    allowedGitHubTeam: envVar("APPROVERS_GH_TEAM"),
+    seedOfTipperAccount: envVar("ACCOUNT_SEED"),
+  };
 
   bot.log.info("Tip bot was loaded!");
 
-  bot.on("issue_comment", (context) => {
+  bot.on("issue_comment.created", async (context) => {
     const tipRequester = context.payload.comment.user.login;
+    const installationId = (
+      await github.getRepoInstallation({
+        owner: context.payload.repository.owner.login,
+        repo: context.payload.repository.name,
+      })
+    ).data.id;
+
+    const octokitInstance = await github.getInstance({
+      authType: "installation",
+      appId: envVar("GITHUB_APP_ID"),
+      installationId: String(installationId),
+      privateKey: envVar("GITHUB_PRIVATE_KEY"),
+    });
 
     const respondOnResult = async (result: string | Error | undefined) => {
       if (result === undefined) {
         return;
       }
 
-      await context.octokit.issues.createComment({
-        owner: context.payload.repository.owner.login,
-        repo: context.payload.repository.name,
-        issue_number: context.payload.issue.number,
-        body: `@${tipRequester} ${result instanceof Error ? `ERROR: ${displayError(result)}` : result}`,
-      });
+      await github.createComment(
+        {
+          owner: context.payload.repository.owner.login,
+          repo: context.payload.repository.name,
+          issue_number: context.payload.issue.number,
+          body: `@${tipRequester} ${result instanceof Error ? `ERROR: ${displayError(result)}` : result}`,
+        },
+        { octokitInstance },
+      );
     };
 
-    void onIssueComment(state, context, tipRequester).then(respondOnResult).catch(respondOnResult);
+    void onIssueComment(state, context.payload, tipRequester, octokitInstance).then(respondOnResult, respondOnResult);
   });
 };
 
+process.env.PRIVATE_KEY = Buffer.from(envVar("PRIVATE_KEY_BASE64"), "base64").toString();
+
+// Aligning environment between Probot and opstooling-integrations
+process.env.GITHUB_APP_ID = process.env.APP_ID;
+process.env.GITHUB_AUTH_TYPE = "app";
+process.env.GITHUB_PRIVATE_KEY = process.env.PRIVATE_KEY;
```

Comment:
Im curious - could it be that we will need several types of auth? how would setting env look like?
i felt several times during review of opstooling integrations and here that it'd maybe be ideal if auth won't be dependent of env vars, but on some other configuration. Either passed by parameters or some config class wrapper. 
Just leaving thoughts, not a request to an action